action/sealedenvelope.go

```diff
@@ -73,12 +80,23 @@ func (sealed *SealedEnvelope) calcHash() (hash.Hash256, error) {
 		}
 		return rlpSignedHash(tx, sealed.evmNetworkID, sealed.Signature())
 	case iotextypes.Encoding_IOTEX_PROTOBUF:
-		return hash.Hash256b(byteutil.Must(proto.Marshal(sealed.Proto()))), nil
+		return hash.Hash256b(byteutil.Must(sealed.serialize())), nil
```

Comment:
see if it's worth to also optimize L44/51

Reply:
The `*iotextypes.ActionCore` struct can't share the memory buf with the `*iotextypes.Action` sync pool. And, `envelopeHash()` is only called in `Sign()` and `VerifySignature()`. So the change wouldn't contribute much efficiency enhancement.